service/src/main/java/io/camunda/service/TenantServices.java

```diff
@@ -65,14 +66,17 @@ public CompletableFuture<TenantRecord> deleteTenant(final long key) {
     return sendBrokerRequest(new BrokerTenantDeleteRequest(key).setTenantKey(key));
   }
 
-  public TenantEntity getByKey(final Long key) {
-    final SearchQueryResult<TenantEntity> result =
-        search(SearchQueryBuilders.tenantSearchQuery().filter(f -> f.key(key)).build());
-    if (result.total() < 1) {
-      throw new NotFoundException(String.format("Tenant with key %d not found", key));
-    } else {
-      return result.items().stream().findFirst().orElseThrow();
-    }
+  public TenantEntity getTenant(final Long tenantKey) {
+    return findTenant(tenantKey)
```

Comment:
The get by key flow should be the following:
1 - search for the tenant by key without any authorization checks
2- if tenant exists apply authorization check (TENANT.READ) on the tenantId
3- return the tenant if it is authorized

you can check this example https://github.com/camunda/camunda/blob/bd52a5afdf32bd5412f49b754e6e6cf62c3c774c/service/src/main/java/io/camunda/service/ProcessDefinitionServices.java#L54-L69

Reply:
@houssain-barouni Wouldn't the search client filter out entities with missing permission (via ProcessDefinitionAuthorizationQueryTransformer)?

Reply:
@houssain-barouni thanks for that link. I'm used to returning 404 instead of 403 from APIs when the client doesn't have read permissions (to prevent resource enumeration), but if that's the pattern we've been using elsewhere we should probably keep it consistent.